pkg/services/ngalert/models/alert_rule.go

```diff
@@ -173,6 +173,19 @@ type AlertRuleWithOptionals struct {
 	HasPause bool
 }
 
+// SortedAlertRules sorts alert rules by AlertRuleGroupKey and then their index.
+type SortedAlertRules []*AlertRule
```

Comment:
We already have this https://github.com/grafana/grafana/blob/f066e8cdcd178b783d0c62959bea593d3c0c30b0/pkg/services/ngalert/models/alert_rule.go#L496-L505
in the very same file. Although, I think the more explicit sorting function is more obvious, we can replace it with this.

Reply:
Thank you! I didn't see this when I wrote that code :)

Reply:
Not sure what's the decision to make here - and sorry if my comment is obvious, but they're technically not `SortedAlertRules` they're more of `SorteableAlertRules` as you'd still need to call sort on them to be sorted.

One more question though, why not include this directly in `ngmodels.AlertRule` is there something I'm missing?

Reply:
I agree, will fix the name! I'm not sure I understand the last question though. How would we include a sorted slice of `[]AlertRule` in `ngmodels.AlertRule`?